core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -54,17 +58,29 @@
    * Factory for {@link FileSystem}.
    */
   class Factory {
+    private static final Logger LOG = LoggerFactory.getLogger(Factory.class);
 
     private Factory() {} // prevent instantiation
 
     public static FileSystem get() {
-      if (Configuration.getBoolean(PropertyKey.USER_LINEAGE_ENABLED)) {
-        return LineageFileSystem.get(FileSystemContext.INSTANCE, LineageContext.INSTANCE);
-      }
-      return BaseFileSystem.get(FileSystemContext.INSTANCE);
+      return get(FileSystemContext.INSTANCE);
     }
 
     public static FileSystem get(FileSystemContext context) {
+      if (LOG.isDebugEnabled()) {
```

Comment:
this will do the logging whenever the user creates a new FileSystem with `FileSystem.get()`, which could happen many times depending on how users structure their programs. The logging contains hundreds of properties, so perhaps we should change this to happen once per JVM.

Reply:
added a flag